File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:

        if head == None or head.next == None:
            return head

        curr = head.next
        prev = head

        while curr != None:
            duplicatesPresent = False
            
            while curr != None and prev.val == curr.val:
                duplicatesPresent = True
                curr = curr.next

            if duplicatesPresent == True:
                prev.next = curr

            if curr == None: break
            else:
                prev = curr
                curr = curr.next

        return head

File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteDuplicates(head *ListNode) *ListNode {
	curr := head
	var prev *ListNode
	for curr != nil {
		prev = curr
		curr = curr.Next
		for curr != nil && prev.Val == curr.Val {
			prev.Next = curr.Next
			curr.Next = nil
			curr = prev.Next
		}
	}
	return head
}